client/src/controllers/BlockController.ts

```diff
@@ -0,0 +1,67 @@
+import { Controller } from '@hotwired/stimulus';
+import { object } from 'prop-types';
+
+/**
+ * Adds the ability to control a block element.
+ *
+ * @example
+ * <div
+ * id="{id}"
+ * data-block
+ * data-controller="w-block"
+ * data-w-block-data-value="{block_json}"
+ * data-w-block-initial-value="{value_json}"
+ * data-w-block-error-value="{error_json}">
+ * </div>
+ */
+
+declare global {
+  interface Window {
+    initBlockWidget?: (id: string) => void; // Declare the initBlockWidget function as optional on the Window interface
+    telepath?: any;
+  }
+}
+
+export class BlockController extends Controller<HTMLElement> {
+  static values = {
+    initial: { type: object, default: {} }, // Initial value for the block
+    data: object, // Unpacked Telepath data
+    error: object, // Block error (optional)
+  };
+
+  declare dataValue: object;
+  declare errorValue: object;
+  declare initialValue: object;
+
+  connect() {
+    const output = window.telepath.unpack(this.dataValue);
+    output.render(
+      this.element,
+      this.element.id,
+      this.initialValue,
+      this.errorValue,
+    );
+    this.dispatch('ready', { detail: { ...output }, cancelable: false });
+  }
+
+  static afterLoad() {
+    window.initBlockWidget = (id: string) => {
```

Comment:
```suggestion
    /**
    * @deprecated RemovedInWagtail70
    */
    window.initBlockWidget = (id: string) => {
```

Let's also flag this as deprecated, it is only being added here to support backwards compatible usage of this global function.